src/server/main_service.cc

```diff
@@ -1101,10 +1101,16 @@ void Service::DispatchManyCommands(absl::Span<CmdArgList> args_list,
 
     // MULTI...EXEC commands need to be collected into a single context, so squashing is not
     // possible
-    const bool is_multi =
+    bool is_multi =
         dfly_cntx->conn_state.exec_info.IsCollecting() || CO::IsTransKind(ArgS(args, 0));
 
-    if (!is_multi && cid != nullptr) {
+    // Generally, executing any multi-transactions (including eval) is not possible because they
+    // migh request a stricter multi mode than non-atomic which is used for squashing.
+    // TODO: By allowing promotion we can potentially execute multiple eval in parallel, which is
+    // very powerful paired with shardlocal eval
```

Comment:
I don't follow this comment. Promotion as in - allow squashing?
In the past you said that squashing doesn't make sense for single shard lua, I think

Reply:
Fixed the comment, made it more detailed